### ISSUE TITLE
Pipeline key match is case-sensitive

### ISSUE THREAD

**Issue by vreuter** [pepkit/peppy#225]
What's are everyone's / anyone's thoughts about making it case-insensitive? Example: right now, defining a `pipeline_args` section of a project config with `WGBS` and `RRBS` as section keys for opt/arg pairs won't match a pipeline interface that names its pipelines `wgbs` and `rrbs`, but obviously the failure to match is silent since the intent can't be inferred.

**Comment by vreuter** [comment]
@nsheff @afrendeiro @michalstolarczyk @aaron-gu @jpsmith5 , others...anyone feel free to chime in

**Comment by aaron-gu** [comment]
Can there be a notification upon failure saying something like "Did you mean to use 'wgbs' instead of 'WGBS'"?

**Comment by vreuter** [comment]
Yeah I think there should be a message regardless; I'd just like to make _match-and-proceed_ be the default, with a notification message saying that's what's going on. That's my preference anyway but wanted to get an little poll of others' thoughts.

**Comment by nsheff** [comment]
I like the match-warn-and-proceed idea.

so, first it matches case sensitive. if it can't find one, it looks case-insensitive and proceeds, with a warning.

**Comment by vreuter** [comment]
My question would be...what fraction of the time would someone *not* want to proceed with, e.g. `rrbs` if they put `RRBS`?

**Comment by nsheff** [comment]
true, but the interface file could potentially be read by other tools, and every that reads it would then have to implement this... or it works with some tools and not others. I think I'd rather force the user to get it right. actually I think I got that philosophy from you @vreuter :smile: 

I don't have strong feelings on it though, as long as there's a clear warning.

**Comment by nsheff** [comment]
Alright I conclude we should do the following:

First match case sensitive. if that fails, match case-insensitive -- if it finds, it just says "did you mean ____?" and does not proceed.

**Comment by vreuter** [comment]
what's the error type?

**Comment by nsheff** [comment]
It should probably be just the same as when you don't have a match, whatever we're doing now.

**Comment by vreuter** [comment]
:ok_hand:
